docs/contracts_api.md

```diff
@@ -0,0 +1,183 @@
+
+**Endpoint URL's:**
+- /contracts
+- /contracts/{contractId}
+
+**Request Path Parameters:**
+- contractId - the specific GUUID id of the contract
+
+**Response:**
+- 200 OK
+```
+{
+    type: Temp Hostel GF,
+    id: 986a2a9e-9eb4-0966-120a-238689e3e265,
+    targetId: 116a2a9e-9eb4-0966-120a-238689e3e265,
+    targetType: asset,
+    startDate: 26/12/2022,
+    endDate: 25/12/2023,
+    renewDate: 25/12/2023,
+    persons: [
+        {
+            id: 986a2a9e-9eb4-0966-120a-238689e3e265,
+            firstname: John,
+            lastname: Doe,
+            email: john@doe.com,
+            type: Hostel manager
+        },
+        {
+            id: 126a2a9e-9eb4-0966-120a-238689e3e269,
+            firstname: Joe,
+            lastname: Soap,
+            email: joe@soap.com,
+            type: Caretaker
+        }
+    ],
+    rates: [
+        {
+            frequency: weekly,
+            value: 85
+        },
+        {
+            frequency: nightly,
+            value: 12
+        }
+    ]
+}
+```
+
+### POST Contracts:
+
+**Endpoint Status**: PROPOSED
+
+**Purpose:**
+Create a new Contract
+
+**Endpoint URL's:**
+/contracts
+
+**Request object:**
+```
+{
+    type: Temp Hostel GF,
+    targetId: 116a2a9e-9eb4-0966-120a-238689e3e265,
+    targetType: asset,
+    startDate: 26/12/2022,
+    endDate: 25/12/2023,
+    renewDate: 25/12/2023,
+    persons: [
+        {
+            id: 986a2a9e-9eb4-0966-120a-238689e3e265,
+            firstname: John,
+            lastname: Doe,
+            email: john@doe.com,
+            type: Hostel manager
+        },
+        {
+            id: 126a2a9e-9eb4-0966-120a-238689e3e269,
+            firstname: Joe,
+            lastname: Soap,
+            email: joe@soap.com,
+            type: Caretaker
+        }
+    ],
+    rates: [
+        {
+            frequency: weekly,
+            value: 85
+        },
+        {
+            frequency: nightly,
+            value: 12
+        }
+    ]
+}
+```
+**Response:**
+- 201 Created
+```
+{
+    type: Temp Hostel GF,
+    targetId: 116a2a9e-9eb4-0966-120a-238689e3e265,
+    targetType: asset,
+    startDate: 26/12/2022,
+    endDate: 25/12/2023,
+    renewDate: 25/12/2023,
+    persons: [
+        {
+            id: 986a2a9e-9eb4-0966-120a-238689e3e265,
+            firstname: John,
+            lastname: Doe,
+            email: john@doe.com,
+            type: Hostel manager
+        },
+        {
+            id: 126a2a9e-9eb4-0966-120a-238689e3e269,
+            firstname: Joe,
+            lastname: Soap,
+            email: joe@soap.com,
+            type: Caretaker
+        }
+    ],
+    rates: [
+        {
+            frequency: weekly,
+            value: 85
+        },
+        {
+            frequency: nightly,
+            value: 12
+        }
+    ]
+}
+```
+
+### PUT Contracts:
+
+**Endpoint Status**: NOT YET PROPOSED
+
+**Purpose:**
+Updates a new Contract
+
+**Endpoint URL's:**
+- /contracts/{contractId}
+
+### DELETE Contracts:
```

Comment:
Under what circumstance do we need the delete endpoint? Bit apprehensive about it.

Reply:
Good point. Again I think this comes back to whether we deem contracts immutable and would we always want full history of contracts.
A question for a combination of data group and officers